core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalH2OBackend.scala

```diff
@@ -38,40 +38,24 @@ import scala.io.Source
 class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Logging with RestApiUtils {
 
   var yarnAppId: Option[String] = None
-  private var externalIP: Option[String] = None
 
   override def init(conf: H2OConf): Array[NodeDesc] = {
     if (conf.isAutoClusterStartUsed) {
-      logInfo("Starting the external H2O cluster on YARN.")
-      val ipPort = launchExternalH2OOnYarn(conf)
-      conf.setH2OCluster(ipPort)
-      val clientIp = ExternalH2OBackend.identifyClientIp(ipPort.split(":")(0))
-      if (clientIp.isDefined && conf.clientIp.isEmpty && conf.clientNetworkMask.isEmpty) {
-        conf.setClientIp(clientIp.get)
-      }
-    } else {
-      val clientIp = ExternalH2OBackend.identifyClientIp(conf.h2oClusterHost.get)
-      if (clientIp.isDefined && conf.clientIp.isEmpty && conf.clientNetworkMask.isEmpty) {
-        conf.setClientIp(clientIp.get)
-      }
-    }
-
-    if (conf.clientIp.isEmpty) {
-      conf.setClientIp(ExternalH2OBackend.getHostname(SparkEnv.get))
+      launchExternalH2OOnYarn(conf)
     }
 
     logInfo("Connecting to external H2O cluster.")
     val nodes = getAndVerifyWorkerNodes(conf)
     if (!RestApiUtils.isRestAPIBased(hc)) {
-      ExternalH2OBackend.startH2OClient(hc, nodes)
+      ExternalH2OBackend.startH2OClient(hc, conf, nodes)
     }
     nodes
   }
 
   override def backendUIInfo: Seq[(String, String)] = {
     Seq(
       ("External backend YARN AppID", yarnAppId),
-      ("External IP", externalIP)
+      ("External IP", hc.getConf.h2oClusterHost)
```

Comment:
it seems to me that `externalIP` https://github.com/h2oai/sparkling-water/pull/1892/files#diff-f5f6b5c818efd05c837b2a930123d96eL118 used to contain also a port. Shouldn't `hc.getConf.h2oCluster` be used here?

Reply:
Good catch, thanks!